File: src/main/java/com/blink/backend/domain/integration/waha/dto/WahaSessionStatus.java
package com.blink.backend.domain.integration.waha.dto;

import com.blink.backend.domain.model.message.WhatsAppStatus;
import lombok.AllArgsConstructor;
import lombok.Getter;

import static com.blink.backend.domain.model.message.WhatsAppStatus.CONNECTED;
import static com.blink.backend.domain.model.message.WhatsAppStatus.DISCONNECTED;
import static com.blink.backend.domain.model.message.WhatsAppStatus.SHUTDOWN;

@Getter
@AllArgsConstructor
public enum WahaSessionStatus {
    STOPPED(SHUTDOWN),
    STARTING(DISCONNECTED),
    SCAN_QR_CODE(DISCONNECTED),
    WORKING(CONNECTED),
    FAILED(SHUTDOWN),
    NOT_FOUND(SHUTDOWN),
    ;

    private final WhatsAppStatus connectionStatus;

    public boolean isShutdown() {
        return connectionStatus == SHUTDOWN;
    }

    public boolean isConnected() {
        return connectionStatus == CONNECTED;
    }
}

File: src/main/java/com/blink/backend/domain/integration/waha/dto/WahaSessionStatus.kt
package com.blink.backend.domain.integration.waha.dto

import com.blink.backend.domain.model.message.WhatsAppStatus

enum class WahaSessionStatus(val connectionStatus: WhatsAppStatus) {
    STOPPED(WhatsAppStatus.SHUTDOWN),
    STARTING(WhatsAppStatus.DISCONNECTED),
    SCAN_QR_CODE(WhatsAppStatus.DISCONNECTED),
    WORKING(WhatsAppStatus.CONNECTED),
    FAILED(WhatsAppStatus.SHUTDOWN),
    NOT_FOUND(WhatsAppStatus.SHUTDOWN),
    ;

    val isShutdown: Boolean
        get() = connectionStatus == WhatsAppStatus.SHUTDOWN

    val isConnected: Boolean
        get() = connectionStatus == WhatsAppStatus.CONNECTED
}